moveit_ros/planning/planning_scene_monitor/src/planning_scene_monitor.cpp

```diff
@@ -780,6 +780,42 @@ bool PlanningSceneMonitor::newPlanningSceneMessage(const moveit_msgs::msg::Plann
   return result;
 }
 
+bool PlanningSceneMonitor::processCollisionObjectMsg(const moveit_msgs::msg::CollisionObject::ConstSharedPtr& object)
+{
+  if (!scene_)
+    return false;
+
+  updateFrameTransforms();
+  {
+    std::unique_lock<std::shared_mutex> ulock(scene_update_mutex_);
+    last_update_time_ = rclcpp::Clock().now();
+    if (!scene_->processCollisionObjectMsg(*object))
+      return false;
+  }
+  triggerSceneUpdateEvent(UPDATE_GEOMETRY);
+  RCLCPP_INFO(logger_, "Published update collision object");
+  return true;
+}
+
+bool PlanningSceneMonitor::processAttachedCollisionObjectMsg(
+    const moveit_msgs::msg::AttachedCollisionObject::ConstSharedPtr& object)
+{
+  if (scene_)
+  {
+    updateFrameTransforms();
+    {
+      std::unique_lock<std::shared_mutex> ulock(scene_update_mutex_);
+      last_update_time_ = rclcpp::Clock().now();
+      if (!scene_->processAttachedCollisionObjectMsg(*object))
+        return false;
+    }
+    triggerSceneUpdateEvent(UPDATE_GEOMETRY);
+    RCLCPP_INFO(logger_, "Published update attached");
+    return true;
+  }
+  return false;
```

Comment:
Small nitpick. I like it a bit better if "correct" return is done at the end.
```suggestion
  if (!scene_)
  {
    return false;
  }

  updateFrameTransforms();
  {
    std::unique_lock<std::shared_mutex> ulock(scene_update_mutex_);
    last_update_time_ = rclcpp::Clock().now();
    if (!scene_->processAttachedCollisionObjectMsg(*object))
      return false;
  }
  triggerSceneUpdateEvent(UPDATE_GEOMETRY);
  RCLCPP_INFO(logger_, "Published update attached");
  return true;
```